lib/data_table.rb

```diff
@@ -467,7 +467,7 @@ def get_resultset_and_count(params)
 
         query = self.query!(params)
         query.offset = params.fetch('skip', 0).to_i
-        query.limit = params.fetch('take', @options[:paging][:pageSize]).to_i
+        query.limit = params.fetch('take', params[:take]).to_i if params[:take]
```

Comment:
move to before this line:
```        
query_limit = nil
unless options.fetch(:no_limit, false)
  query_limit = options.fetch(:limit, 1000)# putting hard limit unless no_limit to prevent issues (not ideal of course)
end
params[:take] = query_limit

query.limit = params.fetch('take', params[:take]).to_i if params[:take]
```

remove from the other methods